text/0031-configuration.md

```diff
@@ -0,0 +1,287 @@
++ Feature name: `configuration`
++ Start date: 2020-05-06
++ RFC PR: [iotaledger/bee-rfcs#31](https://github.com/iotaledger/bee-rfcs/pull/31)
++ Bee issue: [iotaledger/bee#71](https://github.com/iotaledger/bee/issues/71)
+
+# Summary
+
+This RFC proposes a configuration pattern for Bee binary and library crates.
+
+# Motivation
+
+This RFC contains a set of recommendations regarding configuration management in order to ensure consistency across the
+different Bee crates.
+
+# Detailed design
+
+## Libraries
+
+- [serde](https://github.com/serde-rs/serde): the go-to framework for **ser**ializing and **de**serializing Rust data
+  structures efficiently and generically;
+- [toml-rs](https://github.com/alexcrichton/toml-rs): a TOML encoding/decoding library for Rust with
+  serialization/deserialization on top of `serde`;
+
+## Recommendations
+
+With the following recommendations, one can:
+- read a configuration builder from a file;
+- write a configuration to a file;
+- manually override a configuration builder;
+- construct the configuration;
+
+### Configuration Builder
+
+The [Builder](https://rust-lang.github.io/api-guidelines/type-safety.html#c-builder) pattern is very common in Rust when
+it comes to constructing complex objects like configurations.
+
+A configuration builder type should:
+- be named `*ConfigBuilder`;
```

Comment:
Instead of using `*` (interpretable as Deref operator or as regex 0 or many) I would either prefer `...ConfigBuilder` if you want to use a symbol, or - which I prefer most - `MyTypeConfigBuilder`.

Reply:
Hmm not sure about the dots. Would "Should be suffixed with `ConfigBuilder `" be fine ?

Reply:
Let's not speak about crate, I already have some crates with different conf objects inside so this is confusing.

Reply:
okay, yea, rephrasing should also be fine. :+1: